### ISSUE TITLE
css mirroring fail with static files in sub-directory

### ISSUE THREAD

**Issue by chris-miner** [microsoft/vscode-edge-devtools#903]
**Environment (please complete the following information):**
 - OS (*e.g.,* Windows 10):
 - Extension version (*e.g.,* 1.2.3):

**Describe the bug:**

If I put my html, css, and js in a subdirectory of my project folder, updating css doesn't work.  I get the following error:
<img width="462" alt="image" src="https://user-images.githubusercontent.com/94078897/155553542-f2c8b434-ee28-4481-b54b-e78252b3a7b5.png">

**Repro steps:**

1. Create a project containing a single folder titled 'public'
2. Add index.html, script.js, and style.css to the public folder
3. link the script and css files in the html, and include some css to edit later via mirroring
4. create an edge launch configuration with the property `"webRoot": "${workspaceFolder}/public"` and a url port of `5500` (for the live server default)
5. configure live server to vend only the public folder  and start it
6. launch edge
7. open dev tools
8. uncheck a property in one of your rules

**Expected behavior:**

I expected the css property to be updated in my css file.

**Additional context:**

I notice if I leave everything in the project root, and remove the `public` extension on the `webRoot` property everything just works.

Alternatively, If I:
- remove the public extension from webRoot and add it to the launch url
- add a base tag to html to load js and css correctly
- serve the project home directory instead of public css is editable.
then everything also works, but now I have 'public' in my urls.

basically, it seems like the paths have to match.  If the browser fetches /public/style.css then it is okay to have style.css in the public sub-directory.

**Comment by chris-miner** [comment]
Thinking about the paths, made me wonder what happens if I linked the files to my project root.  `ln public/* .`

In that case break points are hit and css is updated.  I'd expect this because now 'something' is finding the right file in the project directory.

**Comment by chris-miner** [comment]
I had a look at #543 and noticed that @mok-liee had included the following in their settings.json
```
"vscode-edge-devtools.pathMapping": {
    "/": "${workspaceFolder}\\frontend"
  },
```
so I tried something similar:
```
"vscode-edge-devtools.pathMapping": {
        "/": "${workspaceFolder}/public"
    },
```

Now I can edit CSS.  But that broke debugging with breakpoints.

**Comment by codepo8** [comment]
Sounds like an enhancement opportunitty to flag this up for people.

**Comment by mliao95** [comment]
@chris-miner 

I was not able to repro this on the latest version of the extension (1.4.5).  Could you try updating to that version and see if you can still repro your issue?  Thanks.

**Comment by chris-miner** [comment]
@mliao95 I can still reproduce my issue.  To make CSS editing work I need to add a workspace setting in `settings.json`:
```
    "vscode-edge-devtools.pathMapping": {
        "/": "${workspaceFolder}/public"
    },
```

**Comment by mliao95** [comment]
@chris-miner Could you provide me with your `launch.json` config that repro'd this bug?

**Comment by mliao95** [comment]
Here is my current setup:
`launch.json`

```
    "configurations": [
        {
            "type": "pwa-msedge",
            "request": "launch",
            "name": "Open index.html",
            "url": "http://127.0.0.1:5500/",
            "webRoot": "${workspaceFolder}/public",
        }
    ]
```

Inside my `settings.json`:
`"liveServer.settings.root": "/public"`

I'm starting the live server and then opening the DevTools through the connection toolbar shortcut:
![image](https://user-images.githubusercontent.com/14304598/157765538-6dfaf144-da62-48dc-adfe-200cbddddbfe.png)

When the DevTools opens, I am able to use css sync without any issue.  Let me know if I am deviating from your repro case here.

**Comment by chris-miner** [comment]
Here's my launch.json:
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "type": "pwa-msedge",
            "request": "launch",
            "name": "Launch Edge",
            "url": "http://localhost:5500/",
            "webRoot": "${workspaceFolder}/public",
            "runtimeArgs": [
                "--headless"
            ],
            "preLaunchTask": "StartServer",
            "postDebugTask": "StopServer"
        },
        {
            "name": "Open Edge DevTools",
            "request": "launch",
            "type": "vscode-edge-devtools.debug",
            "url": "http://localhost:5500/",
            "webRoot": "${workspaceFolder}/public"
        }
    ],
    "compounds": [
        {
            "name": "Start Debug",
            "configurations": [
                "Launch Edge",
                "Open Edge DevTools"
            ]
        }
    ]
}
```

**Comment by chris-miner** [comment]
Here's my settings.json:
```
{
    "liveServer.settings.root": "/public",
    "vscode-edge-devtools.pathMapping": {
        "/": "${workspaceFolder}/public"
    },
    // "vscode-edge-devtools.mirrorEdits": true
}
```

**Comment by mliao95** [comment]
You said that the `pathMapping` fixed it, right?  So should I remove that line to repro the original issue?

**Comment by mliao95** [comment]
Ah I see the issue now. We have already fixed this issue, but won't be public until our next release.  We aim to get the next version out sometime near the end of next week.  Thanks for the patience.

**Comment by mliao95** [comment]
New version released.  Closing issue.
